dashboard/app/assets/javascripts/client_state.js

```diff
@@ -74,20 +70,14 @@ dashboard.clientState.sourceForLevel = function (scriptName, levelId, timestamp)
  * @param {string} source
  */
 dashboard.clientState.writeSourceForLevel = function (scriptName, levelId, timestamp, source) {
-  try {
-    sessionStorage.setItem(createKey(scriptName, levelId, 'source'), JSON.stringify({
-      source: source,
-      timestamp: timestamp
-    }));
-  } catch (e) {
-    if (e.name !== "QuotaExceededError") {
-      throw e;
-    }
-  }
+  sessionStorage.safelySetItem(createKey(scriptName, levelId, 'source'), JSON.stringify({
```

Comment:
As we discussed, let's make this a helper function rather than extending the prototype.